pymc/distributions/multivariate.py

```diff
@@ -1143,73 +1117,115 @@ def _random(self, n, eta, size=1):
                 "too many dimensions. It must yield samples "
                 "with 0 or 1 dimensions. Got {} instead".format(D.ndim - C.ndim - 2)
             )
+
         C *= D[..., :, np.newaxis] * D[..., np.newaxis, :]
         tril_idx = np.tril_indices(n, k=0)
-        return np.linalg.cholesky(C)[..., tril_idx[0], tril_idx[1]]
 
-    def random(self, point=None, size=None):
+        samples = np.linalg.cholesky(C)[..., tril_idx[0], tril_idx[1]]
```

Comment:
The `LKJCorrRV` can be called over here if this particular operation (i.e. `np.linalg.cholesky(C)`) along with `C *= D[..., :, np.newaxis] * D[..., np.newaxis, :]` can be done after we index the `C` array as `C[..., tril_idx[0], tril_idx[1]]`.

If that's the case then sure we could avoid rewriting the code above and directly call `LKJCorrRV.rng_fn` upon these arguments.

Reply:
I am not sure, but worth investigating it